libminifi/include/utils/StringUtils.h

```diff
@@ -17,24 +17,23 @@
 #ifndef LIBMINIFI_INCLUDE_UTILS_STRINGUTILS_H_
 #define LIBMINIFI_INCLUDE_UTILS_STRINGUTILS_H_
 
-#include <string>
-#include <utility>
-#include <iostream>
+#include <algorithm>
 #include <cstring>
 #include <functional>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
 #ifdef WIN32
   #include <cwctype>
   #include <cctype>
 #endif
-#include <algorithm>
-#include <sstream>
-#include <vector>
-#include <map>
-#include <type_traits>
 #include "utils/FailurePolicy.h"
-#include "utils/GeneralUtils.h"
 #include "utils/gsl.h"
-#include "utils/OptionalUtils.h"
 
 #if defined(WIN32) || (__cplusplus >= 201103L && (!defined(__GLIBCXX__) || (__cplusplus >= 201402L) ||  (defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE > 4)))
```

Comment:
I think this is always true now, and it could be simplified.

Reply:
The `HAVE_REGEX_CPP` macro doesn't seem to be used anywhere, so I'm removing the whole block. I'll leave the removal of utils::Regex to a followup: https://issues.apache.org/jira/browse/MINIFICPP-1508